framework/include/vectorpostprocessors/VolumeHistogram.h

```diff
@@ -0,0 +1,69 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef VOLUMEHISTOGRAM_H
+#define VOLUMEHISTOGRAM_H
+
+#include "ElementVectorPostprocessor.h"
+
+class VolumeHistogram;
+
+template<>
+InputParameters validParams<VolumeHistogram>();
+
+/**
+ * Compute a histogram of volume fractions binned according to variable values.
+ * This VectorPostprocessor lets you tabulate the volumes in teh simulation domain
+ * where a given variable has certain values.
+ */
+class VolumeHistogram : public ElementVectorPostprocessor
+{
+public:
+  VolumeHistogram(const InputParameters & parameters);
+
+  virtual void initialize();
+  virtual void execute();
+  virtual void finalize();
+  virtual void threadJoin(const UserObject & y);
```

Comment:
Ah, right, general vs. element vector postprocessors is still a bit muddy to me. Technically those are just user objects, right, so they should work as expected. I'll investigate.